packages/synchronizer/src/__tests__/authenticator.spec.ts

```diff
@@ -0,0 +1,252 @@
+import {ApiHandler} from '../handlers/apiHandler.js';
+import {DeviceFlowHandler} from '../handlers/deviceFlowHandler.js';
+import type {AuthenticatorLoginEvent} from '../utils/authenticator.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+describe('Authenticator Tests', () => {
+  before(async () => {
+    await cleanupTmpConfigDir();
+  });
+
+  afterEach(async () => {
+    await cleanupTmpConfigDir();
+  });
+
+  describe('Initialization', () => {
+    it('should return empty user data on init when no auth file present', async () => {
+      const storagePath = await createTmpConfigDir();
+      const authenticator = createDefaultMonokleAuthenticator(new StorageHandlerAuth(storagePath));
+
+      const user = authenticator.user;
+
+      assert.isFalse(user.isAuthenticated);
+      assert.notExists(user.email);
+      assert.notExists(user.token);
+    });
+
+    it('should return user data on init when auth file present (token method)', async () => {
+      const storagePath = await createTmpConfigDir('token');
+      const authenticator = createDefaultMonokleAuthenticator(new StorageHandlerAuth(storagePath));
+
+      const user = authenticator.user;
+
+      assert.isTrue(user.isAuthenticated);
+      assert.equal(user.email, 'user1@kubeshop.io');
+      assert.equal(user.token, 'USER1_ACCESS_TOKEN');
+      assert.equal(user.data?.auth?.token.token_type, 'access_token');
+    });
+
+    it('should return user data on init when auth file present (device code method)', async () => {
+      const storagePath = await createTmpConfigDir('deviceflow');
+      const authenticator = createDefaultMonokleAuthenticator(new StorageHandlerAuth(storagePath));
+
+      const user = authenticator.user;
+
+      assert.isTrue(user.isAuthenticated);
+      assert.equal(user.email, 'user2@kubeshop.io');
+      assert.equal(user.token, 'USER2_ACCESS_TOKEN');
+      assert.equal(user.data?.auth?.token.token_type, 'bearer');
+    });
+
+    it('should trigger login event on init when auth file present (token method)', async () => {
+      const storagePath = await createTmpConfigDir('token');
+      const authenticator = createDefaultMonokleAuthenticator(new StorageHandlerAuth(storagePath));
+
+      return new Promise(resolve => {
+        authenticator.on('login', (evt: AuthenticatorLoginEvent) => {
+          assert.equal(evt.method, 'token');
+          assert.isTrue(evt.user.isAuthenticated);
+          assert.equal(evt.user.email, 'user1@kubeshop.io');
+          assert.equal(evt.user.token, 'USER1_ACCESS_TOKEN');
+
+          resolve();
+        });
+      });
+    });
+
+    it('should trigger login event on init when auth file present (device code method)', async () => {
+      const storagePath = await createTmpConfigDir('deviceflow');
+      const authenticator = createDefaultMonokleAuthenticator(new StorageHandlerAuth(storagePath));
+
+      return new Promise(resolve => {
```

Comment:
Hint: add a Promise.race with a 300ms timeout that rejects this or you will have to wait a long time when (10-30s) when this fails